### PR TITLE
[Bug] Last cell isn't red on an invalid row &  census overflow

### LINKED ISSUE #469: [Bug] Last cell isn't red on an invalid row
Overflow regression

### LINKED ISSUE #470: [Bug] census overflow
maybe when not enough columns to cause overflow?

screen shot 2019-01-30 at 12 31 53 pm

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #470
closes #469 

last cell isn't red: FIXED
screen shot 2019-01-31 at 1 11 44 pm

overflow was going way past the table: FIXED
screen shot 2019-01-31 at 1 11 32 pm

### CODE REVIEW

File: app/components/display/visual-overflow-table.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default Component.extend({
   },
 
   setOverflow () {
-    let table = this.$()[0];
+    let table = document.querySelector('.ui.striped.celled.table');
```
Comment (James1x0) [inline app/components/display/visual-overflow-table.js]: 👎

File: app/styles/granite/page/census-upload.less [hunk 2]
```diff
@@ -19,7 +19,7 @@
 }
 
 
-.arrow-overflow {
+.arrowOverflow {
```
Comment (James1x0) [inline app/styles/granite/page/census-upload.less]: We don't name classes like this. They should be all lowercase/BEM style.

File: app/styles/granite/page/census-upload.less [hunk 3]
```diff
@@ -43,16 +43,32 @@
...
-  z-index: 3;
+.ui.table {
+  tr.census__highlight-row:nth-child(even) td:last-child {
+     background-color: lighten(@granite-red, 15%);
+     position: relative;
+     z-index: 3;
+  }
+
+  tr:nth-child(even) td:last-child {
+    background: #fafafb;
```
Comment (James1x0) [inline app/styles/granite/page/census-upload.less]: Can we change this to the less variable?

File: app/templates/components/display/visual-overflow-table.hbs [hunk 4]
```diff
@@ -1,2 +1,2 @@
-<div class='arrow-overflow' style={{overflow}} {{action 'quickScroll'}} role="button"></div>
+<div class=arrowOverflow style={{overflow}} {{action 'quickScroll'}} role="button"></div>
```
Comment (James1x0) [inline app/templates/components/display/visual-overflow-table.hbs]: Class should be in double quotes.

File: app/styles/granite/page/census-upload.less [hunk 5]
```diff
@@ -43,16 +43,32 @@
...
+    position: relative;
+    z-index: 3;
```
Comment (James1x0) [inline app/styles/granite/page/census-upload.less]: Does z-index: 3 have any special meaning here? Can it be `1`, `2`?

File: app/styles/granite/page/census-upload.less [hunk 6]
```diff
@@ -43,16 +43,32 @@
   z-index: 3;
 }
 
-.ui.table tr:nth-child(odd) td:last-child {
-  background-color: white;
-  position: relative;
-  z-index: 3;
+.ui.table {
+  tr.census__highlight-row:nth-child(odd) td:last-child {
```
Comment (James1x0) [inline app/styles/granite/page/census-upload.less]: Aren't row colors the same, regardless of even or odd? Can't we combine this selector with the one below?

File: app/templates/components/display/visual-overflow-table.hbs (hunk 4 above)
Comment (JavaLavaMT) [reply app/templates/components/display/visual-overflow-table.hbs]: ✔️

File: app/styles/granite/page/census-upload.less (hunk 5 above)
Comment (JavaLavaMT) [reply app/styles/granite/page/census-upload.less]: changed to 1

File: app/styles/granite/page/census-upload.less (hunk 6 above)
Comment (JavaLavaMT) [reply app/styles/granite/page/census-upload.less]: they are only the same if they are red, else they alternate between white and a darker shade white. when i combined them for the highlight-row my other class takes over and the last cell becomes white or the other darker white. I can keep messing with it to see what i can do

Comment (JavaLavaMT) [reply app/styles/granite/page/census-upload.less]: yes :)

File: app/components/display/visual-overflow-table.js (hunk 1 above)
Comment (JavaLavaMT) [reply app/components/display/visual-overflow-table.js]: ✔️

**Comment by James1x0** [comment]
Are we sure that https://github.com/associatedemployers/granite/pull/471/files#diff-e39c04510f7af4b72323d6ac6870ca04R6 isn't already defined in semantic? The table defaults are likely pulling from a variable.

**Comment by JavaLavaMT** [comment]
census dropdown

File: app/controllers/account/employees/add/census/review.js [hunk 7]
```diff
@@ -51,6 +52,15 @@ export default Controller.extend(ajaxStatus, {
...
   guesses: computed.reads('model.data.0'),
 
+  init () {
+    this._super(...arguments);
+    later(() => {
+      scheduleOnce('afterRender', () => {
+        this.set('$dropdownContext', $('.census__table'));
+      });
+    }, 2000);
+  },
```
Comment (James1x0) [inline app/controllers/account/employees/add/census/review.js]: ✂️

File: app/controllers/account/employees/add/census/review.js [hunk 8]
```diff
@@ -5,6 +5,7 @@ import ajaxStatus from 'granite/mixins/ajax-status';
 import titleCase from 'granite/utils/title-case';
 import { decamelize } from '@ember/string';
 import { states } from 'granite/config/statics';
+import { scheduleOnce, later } from '@ember/runloop';
```
Comment (James1x0) [inline app/controllers/account/employees/add/census/review.js]: ✂️

File: app/templates/account/employees/add/census/review.hbs [hunk 9]
```diff
@@ -53,10 +53,12 @@
               <th>
                 <div class="field">
                   {{#ui-dropdown
-                    class="scrolling census__field-selection"
-                    direction="downward"
+                    class="top right scrolling census__field-selection"
+                    direction="auto"
                     name="guess-column"
                     selected=guesses
+                    keepOnScreen=true
```
Comment (James1x0) [inline app/templates/account/employees/add/census/review.hbs]: I believe this is default

File: app/templates/account/employees/add/census/review.hbs [hunk 10]
```diff
@@ -53,10 +53,12 @@
               <th>
                 <div class="field">
                   {{#ui-dropdown
-                    class="scrolling census__field-selection"
-                    direction="downward"
+                    class="top right scrolling census__field-selection"
+                    direction="auto"
```
Comment (James1x0) [inline app/templates/account/employees/add/census/review.hbs]: Check default for this

Comment (JavaLavaMT) [reply app/templates/account/employees/add/census/review.hbs]: auto is default 🏁

File: app/templates/account/employees/add/census/review.hbs (hunk 9 above)
Comment (JavaLavaMT) [reply app/templates/account/employees/add/census/review.hbs]: yes, you are correct 🏁

File: app/controllers/account/employees/add/census/review.js (hunk 8 above)
Comment (JavaLavaMT) [reply app/controllers/account/employees/add/census/review.js]: 🏁

File: app/controllers/account/employees/add/census/review.js (hunk 7 above)
Comment (JavaLavaMT) [reply app/controllers/account/employees/add/census/review.js]: 🏁

**Pull request merged**